### DISCUSSION TITLE
Expose features through an HTTP API

### DISCUSSION THREAD

**Discussion by tigitz** [GyulyVGC/sniffnet#229] [Ideas]
I'm wondering if exposing all the cool features of SniffNet through an HTTP API could be considered?

I can see myself managing my own servers with sniffnet installed and having a dedicated central UI to manage them all.

It would also open the possibility for everyone to create their own integration and automation.

Finally, it would help users that are not fan of the current UI to build an esthetically improved version. (I admit I'm one of them, not to take away the awesome job you're doing) 

Thanks.

**Comment by GyulyVGC** [comment]
> I'm wondering if exposing all the cool features of SniffNet through an HTTP API could be considered?

It's related to https://github.com/GyulyVGC/sniffnet/issues/57, which is not programmed for the short term.

> Finally, it would help users that are not fan of the current UI to build an esthetically improved version. (I admit I'm one of them, not to take away the awesome job you're doing)

I appreciate your honest comment and I'm also very open to improvements on this subject, please check https://github.com/GyulyVGC/sniffnet/issues/119 if you want to give some feedback about the UI

**Comment by tigitz** [comment]
Noted, thanks for the follow up.

I've checked the UI PR but my idea was more like a complete overhaul.

Dropping the current dark / orange color scheme with something more contrasted and a broader palette. 

Avoid heavy rounded corner. 

Choose a more common font that gives a modern feeling and a less "old school terminal/network" style.

Introduce spacing harmony with carefully chosen space gap between information blocks. 

However these are my own opinions and you're totally free to dismiss them. I'm not a designer per se but I believe these are the basics for a enjoyable interface.

 If you can spare some bucks I recommend https://www.refactoringui.com/ to get a general grasp of a good base design. It's meant to be actionable knowledge for programmers to improve their designs. 

Also https://anthonyhobday.com/sideprojects/saferules/ is very good starting point.

**Comment by GyulyVGC** [comment]
Closing this, please refer to #57 for future updates.
